Deprecate "atomic" SDAG keyword

rbuch opened the pull request:
From @stwhite91's suggestion here: https://github.com/UIUC-PPL/charm/discussions/3682#discussioncomment-4768759. This is the only `.ci` file my grep found that used `atomic`. Are there other examples that use it, @stwhite91?

stwhite91 commented:
Some of the SDAG examples contain .sdag files with atomic in them. Also the wording in the commit msg is backwards, should be 'use serial instead of atomic'.

rbuch commented:
> Some of the SDAG examples (in src/langs/sdag/examples/) contain .sdag files with atomic in them. Also the wording in the commit msg is backwards, should be 'use serial instead of atomic'.

Aha, gotcha, I've added another commit to replace `atomic` in those `.sdag` files. And good catch on the wording, I've corrected that, too.

stwhite91 commented:
I think we should remove atomic from the Charm++ manual section 2.2.4 as well: https://charm.readthedocs.io/en/latest/charm%2B%2B/manual.html#structured-control-flow-structured-dagger

Could maybe leave one note there that atomic is deprecated.
